### PR TITLE
chore(license): update license to Unicode-DFS-2016

### PR DESCRIPTION
**Pull request by srl295**
SPDX: Unicode-DFS-2016

License file same as https://github.com/unicode-org/template-repo/blob/main/LICENSE but with ICU notice at end

### CODE REVIEW

**Comment by srl295** [comment]
> 👉  _note_ (I will switch this to conventional commits, add more scaffolding etc, at the moment this is JUST to review the `LICENSE` file.) 

LICENSE file as received, except I put line breaks in to match the SPDX-verified-file https://github.com/unicode-org/template-repo/blob/main/LICENSE

**Comment by Manishearth** [comment]
Licenses copied over

**Comment by Manishearth** [comment]
I've also added a commit to _disable_ the license header check. We can udpate that in a future PR once this has landed.

**Comment by srl295** [comment]
> I've also added a commit to _disable_ the license header check. We can udpate that in a future PR once this has landed.

OK. Was mainly trying to get confirmation on the contents of the first LICENSE file.

**Comment by Manishearth** [comment]
It seems fine to me!

Also I need to update all the Cargo.tomls, oops

File: Makefile.toml [hunk 1]
```diff
@@ -95,7 +95,8 @@ dependencies = [
 description = "Run all tests for the CI 'tidy' job"
 category = "CI"
 dependencies = [
-    "license-header-check",
+    # Temporarily disabled during license upgrade
+    # "license-header-check",
```
Comment (srl295) [inline Makefile.toml:99]: reinstate this? @Manishearth

Comment (sffc) [reply Makefile.toml:99]: Yes please, reinstate

Comment (srl295) [reply Makefile.toml]: done

**Review by Manishearth — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Pull request merged**